clinica/pipelines/statistics_surface/_model.py

```diff
@@ -0,0 +1,728 @@
+        elif glm_type == "group_comparison":
+            if "*" in design:
+                warnings.warn(
+                    "You included interaction as covariate in your model, "
+                    "please carefully check the format of your tsv files."
+                )
+                return GroupGLMWithInteraction(
+                    design, df, self.feature_label, contrast, **kwargs
+                )
+            return GroupGLM(design, df, self.feature_label, contrast, **kwargs)
+        raise ValueError(
+            f"GLM factory received an unknown GLM type: {glm_type}."
+            f"Only 'correlation' and 'group_comparison' are supported."
+        )
+
+
+class PValueResults:
+    """This class implements a container for raw (uncorrected)
+    P-value results obtained with a GLM model.
+
+    Attributes
+    ----------
+    pvalues: Array of uncorrected P-values.
+    mask: Binary mask.
+    thresh: Threshold.
+    """
+
+    def __init__(
+        self,
+        pvalues: np.ndarray,
+        mask: np.ndarray,
+        threshold: float,
+    ) -> None:
+        self.P = pvalues
+        self.mask = mask
+        self.thresh = threshold
+
+    @classmethod
+    def from_t_statistics(
+        cls,
+        tstats: np.ndarray,
+        df: pd.DataFrame,
+        mask: np.ndarray,
+        threshold: float,
+    ):
+        """Instantiate the class from an array of T-statistics.
+
+        Parameters
+        ----------
+        tstats: Array of T-statistics.
+        df: Subjects DataFrame.
+        mask: Binary mask.
+        threshold: Threshold.
+        """
+        from scipy.stats import t
+
+        return cls(1 - t.cdf(tstats, df), mask, threshold)
+
+    def to_dict(self, jsonable: bool = True):
+        """Returns the `PValueResults` instance in dict format.
+
+        Parameters
+        ----------
+        jsonable: If `True`, the output can be directly written in JSON.
+            Otherwise, it might contain non-serializable objects.
+        """
+        import inspect
+
+        json_dict = dict()
+        for attribute in inspect.getmembers(self):
+            name, value = attribute
+            if not name.startswith("_"):
+                if not inspect.ismethod(value):
+                    if isinstance(value, np.ndarray) and jsonable:
+                        json_dict[name] = value.tolist()
+                    else:
+                        json_dict[name] = value
+        return json_dict
+
+
+class CorrectedPValueResults(PValueResults):
+    """This class implements a container for corrected P-value
+    results obtained with a GLM model.
+
+    Attributes
+    ----------
+
+    """
+
+    def __init__(
+        self,
+        pvalues: np.ndarray,
+        cluster_pvalues: np.ndarray,
+        mask: np.ndarray,
+        threshold: float,
+    ) -> None:
+        super().__init__(pvalues, mask, threshold)
+        self.C = cluster_pvalues
```

Comment:
`self.C`breaks standard Python naming convention. Why not keeping `cluster_pvalues`?

Reply:
See my previous comment.